Compile bug with apple-clang 13

https://github.com/astro-informatics/purify/actions/runs/5334798693/jobs/9667078304
```
/Users/runner/work/purify/purify/cpp/purify/integration.cc:44:25: error: implicit instantiation of undefined template 'std::array<double, 2>'
  std::array<t_real, 2> val{{0., 0.}};
                        ^
/Applications/Xcode_13.2.1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.1.sdk/usr/include/c++/v1/__tuple:219:64: note: template is declared here
template <class _Tp, size_t _Size> struct _LIBCPP_TEMPLATE_VIS array;
                                                               ^
/Users/runner/work/purify/purify/cpp/purify/integration.cc:45:25: error: implicit instantiation of undefined template 'std::array<double, 2>'
  std::array<t_real, 2> err{{0., 0.}};
```

--- a/.github/workflows/cmake.yml
+++ b/.github/workflows/cmake.yml
@@ -62,6 +62,8 @@ jobs:
               cxx: clang++
             - os: macos-11
               mpi: "on"
+            - cxx: clang++
+              omp: "on"
 
     steps:
     - uses: actions/checkout@v3

--- a/cpp/purify/integration.cc
+++ b/cpp/purify/integration.cc
@@ -1,4 +1,5 @@
 #include "purify/integration.h"
+#include <array>
 
 namespace purify {
 namespace integration {

--- a/cpp/purify/integration.cc
+++ b/cpp/purify/integration.cc
@@ -41,8 +41,8 @@ t_complex integrate(const Vector<t_real> &xmin, const Vector<t_real> &xmax,
                     const t_real required_abs_error, const t_real required_rel_error,
                     const t_uint max_evaluations, const method methodtype) {
   assert(xmin.size() == xmax.size());
-  std::array<t_real, 2> val{{0., 0.}};
-  std::array<t_real, 2> err{{0., 0.}};
+  std::array<t_real, 2> val = {0., 0.};
+  std::array<t_real, 2> err = {0., 0.};
   auto wrap_integrand = [](unsigned ndim, const t_real *x, void *fdata, unsigned fdim,
                            double *fval) -> int {
     assert(fdim == 2);

--- a/cpp/purify/integration.cc
+++ b/cpp/purify/integration.cc
@@ -42,8 +42,8 @@ t_complex integrate(const Vector<t_real> &xmin, const Vector<t_real> &xmax,
                     const t_real required_abs_error, const t_real required_rel_error,
                     const t_uint max_evaluations, const method methodtype) {
   assert(xmin.size() == xmax.size());
-  std::array<t_real, 2> val = {0., 0.};
-  std::array<t_real, 2> err = {0., 0.};
+  std::array<t_real, 2> val{0., 0.};
+  std::array<t_real, 2> err{0., 0.};
   auto wrap_integrand = [](unsigned ndim, const t_real *x, void *fdata, unsigned fdim,
                            double *fval) -> int {
     assert(fdim == 2);

